core/src/main/scala/org/bykn/bosatsu/library/Version.scala

```diff
@@ -0,0 +1,195 @@
+package org.bykn.bosatsu.library
+
+import cats.Order
+import cats.parse.Parser
+import cats.parse.Rfc5234.{digit, alpha}
+import cats.parse.Numbers.nonZeroDigit
+import cats.syntax.all._
+
+case class Version(
+  major: Int,
+  minor: Int,
+  patch: Int,
+  preRelease: Option[String],
+  build: Option[String]
+) {
+  override def toString(): String = render
+
```

Comment:
Add a validate method to check that the ints are not negative and pre and build strings are valid.